src/schematools/events/export.py

```diff
@@ -0,0 +1,143 @@
+"""Exporter module.
+
+This module facilitates the export of events from GOB database tables.
+The main objective of this module is to create full events that can be used
+to test the event-importer that is also part of the `schematools.events` package.
+
+The `schematools` cli interface had an entry to use the code in this module.
+"""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import Any, Generator
+
+from geoalchemy2.shape import to_shape
+from json_encoder import json
+from sqlalchemy import Table
+from sqlalchemy.engine import Connection
+
+from schematools.events import metadata
+from schematools.events.factories import tables_factory
+from schematools.types import DatasetSchema, DatasetTableSchema
+from schematools.utils import to_snake_case
+
+
+def fetch_complex_fields_info(dataset_table: DatasetTableSchema) -> dict[str, dict]:
+    """Collect info about complex fields (mainly relations)."""
+    complex_fields = {}
+
+    for field in dataset_table.fields:
+        multi = None
+        properties = {}
+        relation_ds = None
+        try:
+            if (nm_relation := field.nm_relation) is not None:
+                multi = True
+                properties = field["items"]["properties"]
+                relation_ds = nm_relation.split(":")[0]
+            if (relation := field.relation) is not None:
+                multi = False
+                properties = field["properties"]
+                relation_ds = relation.split(":")[0]
+        except KeyError:
+            continue
+
+        if multi is not None:
+            complex_fields[to_snake_case(field.name)] = {
+                "multi": multi,
+                "relation_ds": relation_ds,
+                "identifier_names": dataset_table.identifier,
+                "sub_field_names": [to_snake_case(sf) for sf in properties.keys()],
+            }
+    return complex_fields
+
+
+def collect_nm_embed_rows(
+    dataset_id: str,
+    table_id: str,
+    datasets_lookup: dict[str, DatasetSchema],
+    tables: dict[str, dict[str, Table]],
+    complex_fields_info: dict[str, dict],
```

Comment:
We are still at Python 3.8, hence the parameterized `dict`s should be `Dict`

```python
datasets_lookup: Dict[str, DatasetSchema],
tables: Dict[str, Dict[str, Table]],
complex_fields_info: Dict[str, dict]
```

Reply:
![image](https://user-images.githubusercontent.com/147069/118460721-4117ec00-b6fd-11eb-944f-c1511bc6d918.png)

Reply:
PEP 585

Reply:
Damn, I totally missed that. All those years :facepalm:

Reply:
But MyPy can't yet handle it. No I know why I used the generic classes from `typing` all those years.